x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/textstructure/structurefinder/TextStructure.java

```diff
@@ -598,6 +627,11 @@ public Builder setGrokPattern(String grokPattern) {
             return this;
         }
 
+        public Builder setEcsCompatibility(String ecsCompatibility) {
+            this.ecsCompatibility = TextStructure.getNonNullEcsCompatibilityString(ecsCompatibility);
```

Comment:
The `Builder` class allows `ecsCompatibility` to be `null` (and defaults to that), so you might as well leave conversion of `null` to `disabled` for the constructor of the immutable object.